### ISSUE TITLE
Simplify DepositJob by using SdrClient.async_result interface

### ISSUE THREAD

**Issue by mjgiarlo** [sul-dlss/happy-heron#147]
Blocked by https://github.com/sul-dlss/sdr-client/issues/133

Once that issue is complete, we should be able to simplify DepositJob by allowing sdr-client to manage more of the async mechanics of the sdr-api.

**Comment by mjgiarlo** [comment]
Superseded by #202
